compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2495,11 +2494,19 @@ adjustTileSizesForUnPackOp(mlir::FunctionOpInterface entryPointFn,
   auto pipeline = tInfo.getPassPipeline().getValue();
   auto pipelineConfig = tInfo.getConfiguration();
   if (isLoopPeelingEnabled(entryPointFn)) {
+    // See #16406
     LLVM_DEBUG(KD_DBGS() << "unpack fusion does not work with peeling, falling "
                             "back to non-peeling path");
     pipeline = DispatchLoweringPassPipeline::CPUDoubleTilingExpert;
-    // FIXME: This will overwrite any existing config
-    pipelineConfig = {};
+
+    // Remove the "enable_loop_peeling" attr from pipelineConfig
+    auto enableLoopPeelingAttrName = getEnableLoopPeelingAttrName(rootOp->getContext());
+    auto newAttrs = pipelineConfig.getValue().take_until(
+        [&enableLoopPeelingAttrName](NamedAttribute attr) {
+          return attr.getName() != enableLoopPeelingAttrName;
+        });
```

Comment:
I don't think this is correct. You need to filter out the peeling attribute (it could appear anywhere in the entries list). 
```suggestion
    auto newPipelineConfigEntries = llvm::to_vector(llvm::make_filter_range(
        pipelineConfig.getValue(), [&](NamedAttribute entry) {
          return entry.getName() != enableLoopPeelingAttrName;
        }));
```

Reply:
The `.take_util` logic also looks flipped to me (so even if the pipeline config was just a single enable peeling attribute, it would not remove it).

Reply:
Great catch, thanks Ben! That's a grim reminder to avoid coding past bed-time 😅 Fixed!